controllers/service_controller.go

```diff
@@ -110,6 +115,56 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceReconciler) IsOasDefined(ctx context.Context, service *corev1.Service, log logr.Logger) (bool, string, error) {
+	var oasContent string
+	if oasConfigmapName, ok := service.Annotations[KuadrantDiscoveryAnnotationOASConfigMap]; ok {
+		oasContent, err := r.fetchOpenAPIFromConfigMap(
+			ctx,
+			types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace},
+		)
+		log.V(1).Info("get OAS configmap", "objectKey", types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace}, "error", err)
+		return true, oasContent, err
+	}
+
+	// The annotations for authodiscovery are the following, if not namedPort, it'll use the first one.
+	// discovery.kuadrant.io/oas-path: "/openapi"
+	// discovery.kuadrant.io/oas-port: openapi
+	if oasURLdefined, ok := service.Annotations[KuadrantDiscoveryAnnotationOASPath]; ok {
+		var targetPort int32 = 80
+		if oasNamePort, ok := service.Annotations[KuadrantDiscoveryAnnotationOASNamePort]; ok {
+			for _, port := range service.Spec.Ports {
+				if port.Name == oasNamePort {
+					targetPort = port.Port
+					// TODO: log messsage here
+					break
+				}
+			}
+		} else {
+			if len(service.Spec.Ports) > 0 {
+				targetPort = service.Spec.Ports[0].Port
+			}
+		}
+
+		// This is terrible hack, we should use the same way as Coredns use it, but
```

Comment:
how does CoreDNS use it? How could it be improved?

Reply:
Fixed

Reply:
Sorry, I do not fully get it. We really need to inspect the service type and if
* service type is external name: get the CNAME
* service is headless: get the endpoints
? If that is the case, the controller-runtime should have this already implemented. Maybe we can use it?

I just want to understand what it the best way to do this.

Reply:
I thought that we agreed on private channel on this hack, I'm not sure if you want me to implement the right way, with const, externalName, etc.. 

Regards